### ISSUE TITLE
./x.py dist --install doesn't warn that --install does nothing(?)

### ISSUE THREAD

**Issue by vrusinov** [rust-lang/rust#41855]
I can't seem to be able to build 1.17.0 from source:

```bash
VERSION="1.17.0"
wget https://static.rust-lang.org/dist/rustc-${VERSION}-src.tar.gz
tar xzf rustc-${VERSION}-src.tar.gz
cd rustc-${VERSION}-src
cp $CONFIG .
./x.py build
./x.py dist --install
```

Here's what happens:

```
<...>
+ ./x.py build
<....>
   Compiling libc v0.2.21
   Compiling rustc-serialize v0.3.22
   Compiling getopts v0.2.14
   Compiling gcc v0.3.43
   Compiling filetime v0.1.10
   Compiling num_cpus v0.2.13
   Compiling build_helper v0.1.0 (file:///tmp/build/rustc-1.17.0-src/src/build_helper)
   Compiling cmake v0.1.21
   Compiling toml v0.1.30
   Compiling bootstrap v0.0.0 (file:///tmp/build/rustc-1.17.0-src/src/bootstrap)
    Finished dev [unoptimized] target(s) in 12.51 secs
Build completed unsuccessfully in 0:00:25
+ ./x.py dist --install
    Finished dev [unoptimized] target(s) in 0.0 secs
Build completed unsuccessfully in 0:00:00
```

I.e. it seems like it ignores stuff from config.toml (in $CONFIG of my script) and 'dist --install' just repeats build with default config.

**Comment by Mark-Simulacrum** [comment]
Could you show your config.toml? Otherwise I don't think this is really actionable... not sure what 'ignores stuff' means either -- what do you expect to happen that didn't?

**Comment by vrusinov** [comment]
Sorry, I don't think I have it anymore.

By 'ignores stuff' I meant that `./x.py dist --install` seem to ignore `dist --install` part and just does the same what `build` would do.

**Comment by Mark-Simulacrum** [comment]
Ah, so it looks like `--install` is interpreted as a filter of sort to dist which is the problem here. I'm going to relabel this issue as such.

**Comment by Mark-Simulacrum** [comment]
Though I may have spoken too soon. I don't see --install in the documentation, anyway.

**Comment by bjorn3** [comment]
It now errors with

```
Unrecognized option: 'install'
```

before giving the same help text as `./x.py dist --help`.
